clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -141,6 +141,16 @@ public synchronized void subscribe(Pattern pattern) {
         subscribe(pattern, Optional.empty());
     }
 
+    @Override
+    public void subscribe(SubscriptionPattern pattern, ConsumerRebalanceListener callback) {
+
```

Comment:
It would be very easy to adapt the implementation of `subscribe(Pattern pattern, ConsumerRebalanceListener callback)` for this rather than leaving it blank.

Reply:
Agreed. This is necessary for full coverage of the client-side changes.

Reply:
@AndrewJSchofield, just a question, why do we need to adapt the implementation of `subscribe(Pattern pattern, ConsumerRebalanceListener callback)`  while the `SubscriptionPattern` is to be resolved on the broker, not locally like `Pattern.` If we go that route the `MockConsumer` won't really simulate how the `AsyncConsumer` works

Reply:
I don't mind about the details of the implementation here, but I expect you do need a mocked implemented of this new method in order to complete the PR.